src/app/screens/ConfirmPayment/index.tsx

```diff
@@ -66,7 +60,9 @@ function ConfirmPayment(props: Props) {
       const response = await utils.call(
         "sendPayment",
         { paymentRequest: paymentRequest },
-        { origin: navState.origin }
+        {
+          origin: { ...navState.origin, name: "" }, // name needs to be empty because we do not know the receiver
+        }
```

Comment:
I am having trouble to understand this here.

`navState.origin` can be undefined if in Pop-Up.

But if in Prompt => `navState.origin` exists
And if so, then `navState.origin.name` exists too.
Why reset it then? 
Is the value incorrect for `name` ? 

If so, I would rephrase the comment too:

// always reset name to empty string, because ....

Reply:
You are right, this is confusing. I moved this.  

Context is that we only want `name` to be filled if we know the exact `receiver`. In `confirmPayment` we never know (correct me if I'm wrong @bumi @reneaaron ). So we can leave it out. 
A better place to do this is actually [persistPayments](https://github.com/getAlby/lightning-browser-extension/blob/fix/770_origin-receiver-issue/src/extension/background-script/events/persistPayments.js) because this is where it's saved. Did it there now.

Reply:
I think that's not correct. 
the origin data comes from the website on which the user interacts. So for actions that are triggered from interacting on the website (e.g. webln.sendPayment()) we have metadata. 
if a user manually enters an invoice then we do not have any metadata.  - and we get the receiver name/logo from that website metadata.

why do we never know the receiver in the confirmPayment case? - I think we do in the case described above.

Reply:
Depends on the definition of "receiver". Let's talk.

Reply:
hmm, not sure what definition we need? did you look at the code?